sql-odbc/src/sqlodbc/opensearch_connection.cpp

```diff
@@ -95,6 +95,15 @@ int LIBOPENSEARCH_connect(ConnectionClass *self) {
     rt_opts.crypt.verify_server = (self->connInfo.verify_server == 1);
     rt_opts.crypt.use_ssl = (self->connInfo.use_ssl == 1);
 
+    // Server host url protocol is necessary for sql plugin validation
```

Comment:
I'd like to see `server` assignment and url construction outside of the `if-else` statement.

Seems like a good case for `?:` operator to end up with:
```
auto protocol = use_ssl? "https" : "http";
auto url = std::format("{}://{}", protocol, self->connInfo.server); // or whatever looks good to you
rt_opts.conn.server = url;
```

I'd be ok with it as a one-liner as well.

Reply:
What if server == "https://localhost" ?

Reply:
Right. Didn't notice assignment on line 84. I'll continue this thread in https://github.com/Bit-Quill/opensearch-project-sql/pull/75#discussion_r900547293